app/controllers/google_client_login_controller.rb

```
@@ -0,0 +1,64 @@
+require 'rest_client'
+
+class GoogleClientLoginController < ApplicationController
+  filter_access_to :all
+
+  def index
+    @credentials = GoogleClientLoginCredentials.all
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  def new
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  def create
+    email = params[:email] || (raise ArgumentError, "No email given.")
+    password = params[:password] || (raise ArgumentError, "No password given.")
+
+    if email && password
```

This is somewhat ugly, I'd rather raise an exception at this point. In any case, the else branch should be handled.